### PR TITLE
Add StepFunctionsReadOnlyPolicy

### PR DESCRIPTION
**Pull request by ihalder1**
This policy will allow Lambda functions to list executions of a Step function

*Issue #, if available:*

*Description of changes:*

*Description of how you validated changes:*

*Checklist:*

- [ ] Write/update tests
- [ ] `make pr` passes
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected

*Examples?*

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ihalder1** [comment]
> perform readonly operation in state machine execution

**Comment by ihalder1** [comment]
> # [Codecov](https://codecov.io/gh/aws/serverless-application-model/pull/2028?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=aws) Report
> > Merging [#2028](https://codecov.io/gh/aws/serverless-application-model/pull/2028?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=aws) ([f3ae41b](https://github.com/aws/serverless-application-model/commit/f3ae41b37f6a5a0a5c4258800a6bd4b478ad7703)) into [develop](https://codecov.io/gh/aws/serverless-application-model/commit/54e2a160e182dafcc90399f6af58a1a29dd06f37?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=aws) ([54e2a16](https://github.com/aws/serverless-application-model/commit/54e2a160e182dafcc90399f6af58a1a29dd06f37)) will **increase** coverage by `0.04%`.
> > The diff coverage is `n/a`.
> 
> [Impacted file tree graph](https://codecov.io/gh/aws/serverless-application-model/pull/2028?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=aws)
> 
> ```diff
> @@             Coverage Diff             @@
> ##           develop    #2028      +/-   ##
> ===========================================
> + Coverage    93.58%   93.63%   +0.04%     
> ===========================================
>   Files           90       90              
>   Lines         6080     6080              
>   Branches      1247     1247              
> ===========================================
> + Hits          5690     5693       +3     
> + Misses         182      180       -2     
> + Partials       208      207       -1     
> ```
> 
> [Impacted Files](https://codecov.io/gh/aws/serverless-application-model/pull/2028?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=aws)	Coverage Δ	
> [samtranslator/translator/logical_id_generator.py](https://codecov.io/gh/aws/serverless-application-model/pull/2028/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=aws#diff-c2FtdHJhbnNsYXRvci90cmFuc2xhdG9yL2xvZ2ljYWxfaWRfZ2VuZXJhdG9yLnB5)	`100.00% <0.00%> (+9.09%)`	⬆️
> [Continue to review full report at Codecov](https://codecov.io/gh/aws/serverless-application-model/pull/2028?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=aws).
> 
> > **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=aws)
> > `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> > Powered by [Codecov](https://codecov.io/gh/aws/serverless-application-model/pull/2028?src=pr&el=footer&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=aws). Last update [54e2a16...f3ae41b](https://codecov.io/gh/aws/serverless-application-model/pull/2028?src=pr&el=lastupdated&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=aws). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=aws).

**Review by ihalder1 — COMMENTED** [review]
All good

**Comment by ihalder1** [comment]
reopen

**Comment by ihalder1** [comment]
@szilveszter @pfeilbr @me2resh Can you please review and update the pull request.

**Review by hoffa — APPROVED** [review]

**Comment by ihalder1** [comment]
Thanks @hoffa for approving the pull request. @pfeilbr and @szilveszter this pull request need one more approval, can one of you please review and do the necessary.

**Comment by ihalder1** [comment]
HI @szilveszter please review and approve this change

**Comment by szilveszter** [comment]
@ihalder1 I'm sorry, I'm just another random contributor to this project, and don't have commit rights, so I don't think my review counts.

File: samtranslator/policy_templates_data/policy_templates.json
Comment (wchengru) [inline samtranslator/policy_templates_data/policy_templates.json:1755]: Hi @ihalder1 , the arns for activities, executions and statemachines are different ([doc](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsstepfunctions.html#awsstepfunctions-resources-for-iam-policies)). Can you please add the resources and verify them?

**Review by wchengru — CHANGES_REQUESTED** [review]

**Comment by hawflau** [comment]
Closing the PR as there is no response for a while. Feel free to re-open and ask for approval again after addressing the change request. As @wchengru mentioned, the resource do not apply to all permissions in the policy template.

**Comment by hawflau** [comment]
Closing the PR as there is no response for a while. Feel free to re-open and ask for approval again after addressing the change request. As @wchengru mentioned, the resource do not apply to all permissions in the policy template.

**Pull request closed without merging**
